clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -170,6 +190,20 @@ public synchronized void update(Cluster cluster, long now) {
         this.lastSuccessfulRefreshMs = now;
         this.version += 1;
 
+        if (topicExpiryEnabled) {
```

Comment:
We should update the documentation of `update` to mention that we expire topics if necessary on this call.